src/rust/ide/lib/span-tree/src/node.rs

```diff
@@ -92,43 +151,94 @@ pub fn parent_crumbs(crumbs:&[Crumb]) -> Option<&[Crumb]> {
     crumbs.len().checked_sub(1).map(|new_len| &crumbs[..new_len])
 }
 
+
+
+// ============
 // === Node ===
+// ============
 
 /// SpanTree Node.
 ///
 /// Each node in SpanTree is bound to some span of code, and potentially may have corresponding
 /// AST node.
-#[derive(Clone,Debug,Eq,PartialEq)]
+#[derive(Clone,Debug,Default,Eq,PartialEq)]
```

Comment:
I don't like making span-tree nodes "Default". The span-tree node does not make any sense outside the SpanTree context, and it is very hard to express what in fact the "default nodes" are.

Reply:
As we were chatting via phone, default nodes can be just any node tha t you are adding to spantree while constructing it. A node can default to just insertion point, as this is completely valid default. It simplifies a lot of API. Again, Id love to chat more about it in case it, but in according to rust docs: `Sometimes, you want to fall back to some kind of default value, and don't particularly care what it is.`, which is completely valid here, and simplifies api a lot in different places where we really dont care about this value.